src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -532,7 +532,7 @@ internal static void CheckUnderspecifiedGenericExtension(Symbol extensionMember,
             Debug.Assert(extensionMember.GetIsNewExtensionMember());
 
             NamedTypeSymbol extension = extensionMember.ContainingType;
-            if (extension.ExtensionParameter is not { } extensionParameter || extension.ContainingType.Arity != 0)
+            if (extension.ExtensionParameter is not { } extensionParameter || extension.ContainingType?.Arity != 0)
```

Comment:
>ContainingType [](http://example.com/codeflow?start=88&length=14)

I think we should check for other places in code where we unconditionally accessing `ContainingType` on an extension type.